lib/unison-sqlite/src/Unison/Sqlite/Transaction.hs

```diff
@@ -67,13 +71,33 @@ newtype Transaction a
 
 -- | Run a transaction on the given connection.
 runTransaction :: MonadIO m => Connection -> Transaction a -> m a
-runTransaction conn@(Connection _ _ conn0) (Transaction f) =
-  -- TODO some sensible retry logic
-  liftIO do
-    Sqlite.execute_ conn0 "BEGIN"
-    result <- f conn
-    Sqlite.execute_ conn0 "COMMIT"
+runTransaction conn (Transaction f) = liftIO do
+  uninterruptibleMask \restore -> do
+    Connection.execute_ conn "BEGIN"
+    result <-
+      try (restore (f conn)) >>= \case
+        Left exception -> do
+          rollback
+          case exception of
+            SqliteBusyException ->
+              (`fix` 100_000) \again microseconds -> do
```

Comment:
Please pull out this block to a named function. `fix` is always unnecessarily difficult to read and isn't worth any brevity it affords IMO.

Reply:
Sure